### PR TITLE
Re-oragnize navigation

### PR DESCRIPTION
**Pull request by Karol-Andracki**
# Description

fix of web-navigation (no possibility to copy paste url to diffrent tab or occasionall crashes on page refresh )
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Screenshot(s)

Please attach some screenshot or a movie if provided changes affects UI.

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [ ] Add correct label to your pull request
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/hooks/navigation/useScreenTracker.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { useCallback, useRef } from 'react'
 
 import { navigationRef } from '~utils'
 
-type Callback = (currentRouteName: string) => Promise<void> | undefined
+type Callback = (currentRouteName: string) => Promise<void> | void | undefined
```
Comment (Karol-Andracki) [inline src/hooks/navigation/useScreenTracker.ts]: rollback this change

File: src/services/NotificationService.ts [hunk 2]
```diff
@@ -25,12 +25,11 @@ export const registerForPushNotificationsAsync = async () => {
       finalStatus = status
     }
     if (finalStatus !== 'granted') {
-      alert('Failed to get push token for push notification!')
+      console.log('Failed to get push token for push notification!')
       throw new Error('Failed to get push token for push notification!')
```
Comment (vercia) [inline src/services/NotificationService.ts:29]: after starting the project I can't do anything - if I comment this line the app on web starts working

File: src/screens/TypographyScreen.tsx [hunk 3]
```diff
@@ -31,7 +33,12 @@ export const TypographyScreen = (): JSX.Element => {
             Investigate the issue about using `useCallback` on `onChange`
             https://github.com/adobe/react-spectrum/issues/2320
           */}
-          <Switch mx={4} my={8} value={colorMode === 'dark'} onChange={toggleColorMode} />
+          <Switch
+            mx={4}
+            my={8}
+            value={colorScheme === 'dark'}
+            onChange={() => setColorSchemeSetting(colorScheme === 'dark' ? 'light' : 'dark')}
```
Comment (vercia) [inline src/screens/TypographyScreen.tsx:40]: onChange doesn't work on web, but if we change this function to onToggle - it starts working - on web and mobile

File: src/screens/ApplicationInfoScreen.tsx [hunk 4]
```diff
@@ -1,15 +1,48 @@
...
+        await Notifications.getExpoPushTokenAsync(
+          !isExpoGo
+            ? {
+                experienceId,
+              }
+            : {}
+        )
+      ).data
+
+      console.log(token)
```
Comment (vercia) [inline src/screens/ApplicationInfoScreen.tsx:34]: remove this

File: src/hooks/useNotificationSetup.ts [hunk 5]
```diff
@@ -36,32 +35,48 @@ export const useNotificationSetup = (opts?: Options) => {
...
+
+      if (!authContextRef.current?.isSignedIn && isAuthorizedPath) {
         AsyncStorage.setItem(ASYNC_STORAGE_KEYS.NEXT_DEEP_LINK, deeplinkPath)
         finalDeeplink = '/sign-in'
       }
 
       const addSlash = !finalDeeplink.startsWith('/')
       finalDeeplink = `${addSlash ? '/' : ''}${finalDeeplink}`
 
+      console.log(finalDeeplink)
```
Comment (vercia) [inline src/hooks/useNotificationSetup.ts:47]: remove this

**Pull request merged**